aggregator/counter_elem.gen.go

```diff
@@ -274,8 +270,15 @@ func (e *CounterElem) Close() {
 	e.closed = true
 	e.id = nil
 	e.parsedPipeline = parsedPipeline{}
+	e.writeForwardedMetricFn = nil
+	e.onForwardedAggregationWrittenFn = nil
+	for idx := range e.cachedSourceSets {
```

Comment:
Do you need to clear `e.cachedSourceSets ` if you are setting it to nil below?

Reply:
yeah just for precaution in case the object still gets scanned for some reason.